service/src/java/org/apache/hive/service/auth/HiveAuthFactory.java

```diff
@@ -123,7 +123,7 @@ public HiveAuthFactory(HiveConf conf) throws TTransportException {
     return saslProps;
   }
 
-  public TTransportFactory getAuthTransFactory() throws LoginException {
+  public TTransportFactory getAuthTransFactory(String hiveHost) throws LoginException {
```

Comment:
hiveHost is the host name of HS2. We could either automatically detect this or we could rename this to "validDomain" as we really just specify here, what we use to check against with incoming requests.....

Reply:
As per the latest approach, the trusted domain will be configured and thus we do not rely on hivehost to know the trusted domain.